pkg/pki/pkcs7/pkcs7.go

```diff
@@ -211,5 +213,31 @@ func (k PublicKey) EmailAddresses() []string {
 
 // Subjects implements the pki.PublicKey interface
 func (k PublicKey) Subjects() []string {
-	return k.EmailAddresses()
+	// combine identities in the subject and SANs
+	identities := k.EmailAddresses()
+	cert, err := x509.ParseCertificate(k.rawCert)
+	if err != nil {
+		// This should not happen from a valid PublicKey, but fail gracefully.
+		return identities
+	}
+	identities = append(identities, px509.GetSubjectAlternateNames(cert)...)
+	return identities
+}
+
+// Identities implements the pki.PublicKey interface
+func (k PublicKey) Identities() ([]string, error) {
+	// pkcs7 structure may contain both a key and certificate chain
+	var identities []string
+	if len(k.certs) > 0 {
+		cert := k.certs[0]
+		pem, err := cryptoutils.MarshalPublicKeyToPEM(cert.PublicKey)
```

Comment:
for this you can take `k.key` too, which is `k.certs[0].PublicKey`, so I'm not sure you need this if block.

(get k.key marshalled PEM, and then add k.Subjects()`)

Reply:
cleaned up!